### PR TITLE
Screen mirror visualization

### PR DESCRIPTION
**Pull request by StijnOostdam**
Added visualization of different screen mirror algorithms

### CODE REVIEW

**Comment by bartdebever** [comment]
Visualizing Ambilight does nothing, as intended. Yet it doesn't feel natural as the button does nothing.

**Comment by bartdebever** [comment]
When visualizing fit and then visualizing stretch without closing fit, it gives an odd effect where the two merge.

**Comment by bartdebever** [comment]
image
The triangles have this weird uneven gap in it, check the middle lights and the left and the right

**Comment by bartdebever** [comment]
Either way, code looks solid and function is as expected.

File: Nanoleaf-wpf/ViewModels/OptionsViewModel.cs [hunk 1]
```diff
@@ -101,7 +101,7 @@ public string SelectedDevice
                 SelectedMonitor = MonitorPerDevice[_selectedDevice];
 
                 var screenMirrorAlgorithm = AlgorithmPerDevice[_selectedDevice];
-                SelectedScreenMirrorAlgorithm = _screenMirrorAlgorithmMapping.FirstOrDefault(map => map.Value == screenMirrorAlgorithm).Key;
+                SelectedScreenMirrorAlgorithm = ScreenMirrorAlgorithmMapping.FirstOrDefault(map => map.Value == screenMirrorAlgorithm).Key;
 
                 //_parent.SelectedDeviceChanged(); Will be used later if we visualize the algorithms
```
Comment (bartdebever) [inline Nanoleaf-wpf/ViewModels/OptionsViewModel.cs]: Comment no longer applies

File: Nanoleaf-wpf/Views/Options/OptionsWindow.xaml.cs [hunk 2]
```diff
@@ -77,6 +79,27 @@ private void Cancel_Click(object sender, RoutedEventArgs e)
             Close();
         }
 
+        private void Visualize_Click(object sender, RoutedEventArgs e)
+        {
+            var screenMirrorAlgorithm = OptionsViewModel.ScreenMirrorAlgorithmMapping[OptionsViewModel.SelectedScreenMirrorAlgorithm];
+            
+            //Currently, only these 2 alorithm can be visualized
+            if (screenMirrorAlgorithm == ScreenMirrorAlgorithm.ScreenMirrorFit || screenMirrorAlgorithm == ScreenMirrorAlgorithm.ScreenMirrorStretch)
```
Comment (bartdebever) [inline Nanoleaf-wpf/Views/Options/OptionsWindow.xaml.cs]: and here would be the point to give a popup or such for ambilight not being able to display

**Comment by StijnOostdam** [comment]
- Disabled visualize button when ambilight is selected
- Only 1 visualization window can be opened
- Added spacing problem to project as approved task

**Pull request merged**